store/src/java/com/zimbra/cs/redolog/op/CreateSmartFolder.java

```diff
@@ -0,0 +1,70 @@
+package com.zimbra.cs.redolog.op;
+
+import java.io.IOException;
+
+import com.zimbra.cs.mailbox.MailServiceException;
+import com.zimbra.cs.mailbox.Mailbox;
+import com.zimbra.cs.mailbox.MailboxManager;
+import com.zimbra.cs.mailbox.MailboxOperation;
+import com.zimbra.cs.redolog.RedoLogInput;
+import com.zimbra.cs.redolog.RedoLogOutput;
+
+public class CreateSmartFolder extends RedoableOp {
+
+    private int mId;
+    private String mName;
+
+    public CreateSmartFolder() {
+        super(MailboxOperation.CreateSmartFolder);
+        mId = UNKNOWN_ID;
+    }
+
+    public CreateSmartFolder(int mailboxId, String name) {
+        this();
+        setMailboxId(mailboxId);
+        mId = UNKNOWN_ID;
+        mName = name != null ? name : "";
```

Comment:
Are we OK with having an empty name?  Inclined to think should throw an Exception.

Reply:
Agreed, good point.

Reply:
Handling this in the `Mailbox::createSmartFolder` call, since a RedoOp with an empty name shouldn't be created in the first place